pkg/observ/observ.go

```diff
@@ -17,16 +18,35 @@ type observabilityContext struct {
 	spanCtx context.Context
 }
 
-// RegisterTraceExporter returns a jaeger exporter for exporting traces to opencensus.
+// RegisterExporter determines the type of TraceExporter service for exporting traces from opencensus
+// User can choose from multiple tracing services (datadog, jaegar)
+// RegisterExporter returns the 'Flush' function for that particular tracing service
+// TODO: Extend beyond jaeger and datadog
+func RegisterExporter(TraceExporter, URL, service, ENV string) (func(), error) {
+	const op errors.Op = "RegisterExporter"
+	switch TraceExporter {
+	case "jaeger":
+		ex, err := registerJaegerExporter(URL, service, ENV)
+		return ex.Flush, errors.E(op, err)
```

Comment:
like @manugupt1 said, `errors.E` returns a non nil error. Therefore, we should either check the err and return nil, or just have that happen inside the registerX funcs and here you just do `return registerX(..)`

Reply:
@marwan-at-work  Should I change it to this?
```
switch TraceExporter {
	case "jaeger":
		ex, err := registerJaegerExporter(URL, service, ENV)
		if err != nil {
			return nil, errors.E(op, err)
		}
		return ex.Flush, nil
```

Reply:
Updated!!